### DISCUSSION TITLE
Phased exposure - iterations

### DISCUSSION THREAD

**Discussion by Hazoooka** [sn4k3/UVtools#876] [Q&A]
Hi Tiago.

Writing here as there's no 'Bug' section, at least I can not find it.

Phased exposures, iterations do not work properly as it was in Double Exposures.
I want to wipe the contents of additional passes, erase the slices, make them black.
I am setting the normal Iterations for the additional passes to 800 px, but nothing changes, the slices are as before instead of being black.
Before the slices turned black, now they are unchanged.

Tested with:
3.15.1 - Double Exposure
4.0.4  - the last Double Exposure release
4.0.5  - the first Phased Exposure
4.3.2. - the latest release

All Double Exposures worked properly, iterations 800px wiped out the contents of the additional slices.
All Phased Exposures tested didn't work. Setting iterations to 800px do not change a thing.

Kind regards,
Witold Hazuka.

**Comment by sn4k3** [comment]
Phased exposures have a safe check when the resulting erode gives a empty layer, it ignore that layer.
This is the wrong tool to use if you want to wipe a layer.
If you looking for any custom strategy you will have to use scripting, that way you can do whatever you want with layers.

**Comment by Hazoooka** [comment]
Would it be possible to change it the way it was, to be able to wipe the layer?
Wiping the layers with erosion is the whole point of additional exposures.
This is what makes Phased Exposures really usable.

[Edit]
Wiping the slice is what makes it possible to replace it fairly easily with different one to expose different parts of the model shorter or longer.

I am not a programmer, do not know how to write a script.
The new 'safe' in iterations makes it much much harder to work with.

Kind regards,
Witold

**Reply by sn4k3** [reply]
No, it wont be changed.

**Comment by Hazoooka** [comment]
I have had a long break from 3D printing, unintentionally, I couldn't check right away the new multi exposure you have made.

**Comment by Hazoooka** [comment]
Than this tool is unusable.
Can you share the reason why?

**Comment by Hazoooka** [comment]
Could you make it as an option?

**Reply by piotao** [reply]
@Hazoooka maybe scripting is something you may want to consider. Might be easier to conquer? Other that that, it may me this case: 
https://xkcd.com/1172/

**Comment by Hazoooka** [comment]
Hi Pio Tao, probrbly this could be helpful if only I knew how to start.
We talked about this, I am not a programmer, I’m a programming idiot, I  don’t have a talent for it.
I can start learning from scratch of course, but the life goes on, I have projects to complete.
